diff --git a/tests/sidetrail/working/s2n-cbc/utils/s2n_safety.c b/tests/sidetrail/working/s2n-cbc/utils/s2n_safety.c
index 3af0262..261285a 100644
--- a/tests/sidetrail/working/s2n-cbc/utils/s2n_safety.c
+++ b/tests/sidetrail/working/s2n-cbc/utils/s2n_safety.c
@@ -55,8 +55,8 @@ pid_t s2n_actual_getpid()
  */
 int s2n_constant_time_equals(const uint8_t * a, const uint8_t * b, uint32_t len)
 {
-    S2N_PUBLIC_INPUT(a);
-    S2N_PUBLIC_INPUT(b);
+  //S2N_PUBLIC_INPUT(a);
+  //S2N_PUBLIC_INPUT(b);
     S2N_PUBLIC_INPUT(len);

     uint8_t xor = 0;
