diff --git a/tls/s2n_record_read_composite.c b/tls/s2n_record_read_composite.c
--- a/tls/s2n_record_read_composite.c
+++ b/tls/s2n_record_read_composite.c
@@ -25,6 +25,8 @@
 #include "utils/s2n_blob.h"
 #include "utils/s2n_safety.h"
 
+extern int g_padding_length;
+
 int s2n_record_parse_composite(
         const struct s2n_cipher_suite *cipher_suite,
         struct s2n_connection *conn,
@@ -84,6 +86,8 @@
 
     /* Subtract the padding length */
     POSIX_ENSURE_GT(en.size, 0);
+    //After hmac verification padding_length is declassified
+    en.data[en.size - 1] = g_padding_length;
     uint32_t out = 0;
     POSIX_GUARD(s2n_sub_overflow(payload_length, en.data[en.size - 1] + 1, &out));
     payload_length = out;
