diff --git a/tls/s2n_handshake_io.c b/tls/s2n_handshake_io.c
index 60f1f38d..274de930 100755
--- a/tls/s2n_handshake_io.c
+++ b/tls/s2n_handshake_io.c
@@ -888,6 +888,7 @@ static int s2n_advance_message(struct s2n_connection *conn)
         if (s2n_connection_is_managed_corked(conn)) {
             /* Set TCP_CORK/NOPUSH */
             POSIX_GUARD(s2n_socket_write_cork(conn));
+            POSIX_GUARD(s2n_socket_write_uncork(conn));
         }
 
         return S2N_SUCCESS;
