diff -r -u s2n/crypto/s2n_hmac.c s2n_break/crypto/s2n_hmac.c
--- s2n/crypto/s2n_hmac.c	2016-05-23 09:59:56.600873694 -0700
+++ s2n_break/crypto/s2n_hmac.c	2016-05-27 16:12:24.278698053 -0700
@@ -265,6 +265,6 @@ int s2n_hmac_update(struct s2n_hmac_state *state, const void *in, uint32_t size)
      * input. On some platforms, including Intel, the operation can take a
      * smaller number of cycles if the input is "small".
      */
-    const uint32_t HIGHEST_32_BIT = 4294949760;
+    const uint32_t HIGHEST_32_BIT = 4294949761;
     POSIX_ENSURE(size <= (UINT32_MAX - HIGHEST_32_BIT), S2N_ERR_INTEGER_OVERFLOW);
     uint32_t value = (HIGHEST_32_BIT + size) % state->hash_block_size;
Binary files s2n/.git/index and s2n_break/.git/index differ
Only in s2n/tests/saw: bad_magic_mod.patch
