diff --git a/tls/s2n_handshake_io.c b/tls/s2n_handshake_io.c
index d7c84d0..c434db1 100755
--- a/tls/s2n_handshake_io.c
+++ b/tls/s2n_handshake_io.c
@@ -137,7 +137,7 @@ static message_type_t handshakes[128][16] = {
             APPLICATION_DATA
     },
 
-    [NEGOTIATED | FULL_HANDSHAKE | PERFECT_FORWARD_SECRECY | WITH_SESSION_TICKET ] = {
+    [NEGOTIATED | PERFECT_FORWARD_SECRECY | WITH_SESSION_TICKET ] = {
             CLIENT_HELLO, SERVER_SESSION_LOOKUP,
             SERVER_HELLO, SERVER_CERT, SERVER_KEY, SERVER_HELLO_DONE,
             CLIENT_KEY, CLIENT_CHANGE_CIPHER_SPEC, CLIENT_FINISHED,
@@ -145,7 +145,7 @@ static message_type_t handshakes[128][16] = {
             APPLICATION_DATA
     },
 
-    [NEGOTIATED | FULL_HANDSHAKE | OCSP_STATUS ] ={
+    [NEGOTIATED | OCSP_STATUS ] ={
             CLIENT_HELLO, SERVER_SESSION_LOOKUP,
             SERVER_HELLO, SERVER_CERT, SERVER_CERT_STATUS, SERVER_HELLO_DONE,
             CLIENT_KEY, CLIENT_CHANGE_CIPHER_SPEC, CLIENT_FINISHED,
