diff --git a/utils/s2n_safety.c b/utils/s2n_safety.c
index ae8e5783..cc06a2d0 100644
--- a/utils/s2n_safety.c
+++ b/utils/s2n_safety.c
@@ -57,9 +57,6 @@ s2n_constant_time_equals
  */
 bool s2n_constant_time_equals(const uint8_t * a, const uint8_t * b, const uint32_t len)
 {
-    S2N_PUBLIC_INPUT(a);
-    S2N_PUBLIC_INPUT(b);
-    S2N_PUBLIC_INPUT(len);

     /* if len is 0, they're always going to be equal */
     if (len == 0) {
         return true;
     }
@@ -90,10 +87,6 @@ s2n_constant_time_copy_or_dont
  */
 int s2n_constant_time_copy_or_dont(uint8_t * dest, const uint8_t * src, uint32_t len, uint8_t dont)
 {
-    S2N_PUBLIC_INPUT(dest);
-    S2N_PUBLIC_INPUT(src);
-    S2N_PUBLIC_INPUT(len);
-
     uint8_t mask = (((0xFFFF & dont) - 1) >> 8) & 0xFF;

     /* dont = 0 : mask = 0xff */
     /* dont > 0 : mask = 0x00 */
