diff --git a/tls/s2n_record_read_cbc.c b/tls/s2n_record_read_cbc.c
index bb06523..60a2eed 100644
--- a/tls/s2n_record_read_cbc.c
+++ b/tls/s2n_record_read_cbc.c
@@ -30,6 +30,8 @@
 #include "utils/s2n_safety.h"
 #include "utils/s2n_blob.h"
 
+extern int g_padding_length;
+
 int s2n_record_parse_cbc(
     const struct s2n_cipher_suite *cipher_suite,
     struct s2n_connection *conn,
@@ -86,6 +88,8 @@ int s2n_record_parse_cbc(
 
     /* Subtract the padding length */
     POSIX_ENSURE_GT(en.size, 0);
+    //After hmac verification padding_length is declassified
+    en.data[en.size - 1] = g_padding_length;
     uint32_t out = 0;
     POSIX_GUARD(s2n_sub_overflow(payload_length, en.data[en.size - 1] + 1, &out));
     payload_length = out;
@@ -107,6 +111,7 @@ int s2n_record_parse_cbc(
 
     /* Padding */
     if (s2n_verify_cbc(conn, mac, &en) < 0) {
+        __VERIFIER_assume(0);
         POSIX_GUARD(s2n_stuffer_wipe(&conn->in));
         POSIX_BAIL(S2N_ERR_BAD_MESSAGE);
     }
